src/components/Intro/Index.tsx

```diff
@@ -1,140 +1,216 @@
-        const light1y = state.pointer.y * size.height / 250
-
-        setLight1x(light1x)
-        setLight1y(light1y)
-
-        const light2x = -Math.sin(timer * 0.0001) * 6
-        const light2y = -Math.cos(timer * 0.0004) * 3;
-
-        setLight2x(light2x)
-        setLight2y(light2y)
-
-        setLightIntensity(timer*1.0001)
-    })
-
-
-    return (
-        <>
-            <spotLight position={[light2y, -light2x, 2]} angle={0.3} intensity={Math.min(0.3, lightIntensity)}/>
-            <pointLight position={[0, 0, 10]} intensity={Math.min(0.8, lightIntensity)} distance={20} decay={1}/>
-            <pointLight position={[light1x, light1y, 2]} intensity={Math.min(2.0, lightIntensity)} distance={3} decay={0.4}/>
-
-            <Text3D
-                visible={true}
-                rotation={[0, 0, 0]}
-                position={[-3, -3, 0]}
-                font={'/custom-fonts/sprout-logo/sprout-logo-glyph.json'}
-                curveSegments={3}
-                bevelEnabled
-                bevelSize={0.1}
-                bevelThickness={0.1}
-                height={0.2}
-                lineHeight={0.5}
-                letterSpacing={0.4}
-                size={4.5}>
-                A
-                <meshStandardMaterial color={"grey"}/>
-                {
-                    <AsciiRenderer fgColor="white" bgColor="#0000"/>
-                }
-            </Text3D>
-        </>
-    )
+import { TextGeometry } from "three/examples/jsm/geometries/TextGeometry";
+
+import { AsciiEffect } from "./AsciiEffect.js";
+import { Canvas, extend, useFrame, useThree } from "@react-three/fiber";
+import { Text3D } from "@react-three/drei";
+import {
+  useCallback,
+  useEffect,
+  useLayoutEffect,
+  useMemo,
+  useRef,
+  useState,
+  useSyncExternalStore,
+} from "react";
+
+extend({ TextGeometry });
+
+const asciiRenderOptions = {
+  fgColor: "rgb(255,255,255,0)",
+  bgColor: "rgb(0,0,0,0)",
+};
+
+const BASE_RESOLUTION = 0.2;
+const xlarge = ["(min-width: 1280px)", 0.12] as const;
+const large = ["(min-width: 1024px)", 0.15] as const;
+const medium = ["(min-width: 768px)", 0.17] as const;
+
+const resolutions = [xlarge, large, medium].map(
+  ([mq, res]) => [window.matchMedia(mq), res] as const
+);
+
+const useResolution = () => {
+  const subscribe = useCallback((callback: VoidFunction) => {
+    resolutions.forEach(([mq]) => {
+      // if ("addEventListener" in mq) {
+      mq.addEventListener("change", callback);
+      // } else {
+      // TODO: @ts-expect-error up to safari 13 compat
+      // mq.addListener(callback);
+      // }
+    });
+
+    return () => {
+      resolutions.forEach(([mq]) => {
+        // if ("removeEventListener" in mq) {
+        mq.removeEventListener("change", callback);
+        // } else {
+        // TODO: @ts-expect-error up to safari 13 compat
+        // mq.removeListener(callback);
+        // }
+      });
+    };
+  }, []);
+
+  return useSyncExternalStore(subscribe, () => {
```

Comment:
The MediaQuery Object is pretty much an external stateful store. So we can hook into it with this new React 18 hook 😄 or rather, to 3 of them all at once!